File: client/src/types/prediction.ts
export interface PredictionJob {
  id: number;
  player_id: number;
  status: "pending" | "running" | "completed" | "failed";
  result: {
    predicted_scoring_average: number;
    features_used: Record<string, number>;
    model: string;
  } | null;
  error_message: string | null;
  created_at: string;
  completed_at: string | null;
}

File: server/app/schemas/prediction.py
from datetime import datetime

from pydantic import BaseModel


class PredictionJobCreate(BaseModel):
    player_id: int


class PredictionJobOut(BaseModel):
    id: int
    player_id: int
    status: str
    result: dict | None = None
    error_message: str | None = None
    created_at: datetime
    completed_at: datetime | None = None

    model_config = {"from_attributes": True}
